daemon/algod/api/spec/v2/model.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+// Package v2 defines models exposed by algod rest api
+package v2
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+)
+
+// AccountResourceModel used to encode AccountResource
+type AccountResourceModel struct {
```

Comment:
I think that we *should* have a separate object for the model; that would be a better long-term approach if we want to retain backward compatibility. That being said, if you feel that we need a "trigger" to verify that these contains the same fields, it would be fine to create a unit test that uses reflection to verify that.

Reply:
I was trying to avoid more response types, but it sounds like it is better to have more response types.

Reply:
Addressing in https://github.com/algorand/go-algorand/pull/3679